autoload/go/term.vim

```diff
@@ -30,49 +30,73 @@ function! go#term#newmode(bang, cmd, errorformat, mode) abort
 
   execute l:cd . fnameescape(expand("%:p:h"))
 
-  execute l:mode . ' __go_term__'
-
-  setlocal filetype=goterm
-  setlocal bufhidden=delete
-  setlocal winfixheight
-  setlocal noswapfile
-  setlocal nobuflisted
-
-  " explicitly bind callbacks to state so that within them, self will always
-  " refer to state. See :help Partial for more information.
-  "
-  " Don't set an on_stderr, because it will be passed the same data as
-  " on_stdout. See https://github.com/neovim/neovim/issues/2836
-  let l:job = {
-        \ 'on_stdout': function('s:on_stdout', [], state),
-        \ 'on_exit' : function('s:on_exit', [], state),
-      \ }
-
-  let l:state.id = termopen(a:cmd, l:job)
-  let l:state.termwinid = win_getid(winnr())
-
-  execute l:cd . fnameescape(l:dir)
-
-  " resize new term if needed.
   let l:height = go#config#TermHeight()
   let l:width = go#config#TermWidth()
 
-  " Adjust the window width or height depending on whether it's a vertical or
-  " horizontal split.
-  if l:mode =~ "vertical" || l:mode =~ "vsplit" || l:mode =~ "vnew"
-    exe 'vertical resize ' . l:width
-  elseif mode =~ "split" || mode =~ "new"
-    exe 'resize ' . l:height
-  endif
+  " setup job for nvim
+  if has('nvim')
+    execute l:mode . ' __go_term__'
+    setlocal filetype=goterm
+    setlocal bufhidden=delete
+    setlocal winfixheight
+    setlocal noswapfile
+    setlocal nobuflisted
+
+    " explicitly bind callbacks to state so that within them, self will always
+    " refer to state. See :help Partial for more information.
+    "
+    " Don't set an on_stderr, because it will be passed the same data as
+    " on_stdout. See https://github.com/neovim/neovim/issues/2836
+    let l:job = {
+          \ 'on_stdout': function('s:on_stdout', [], state),
+          \ 'on_exit' : function('s:on_exit', [], state),
+        \ }
+    let l:state.id = termopen(a:cmd, l:job)
+    let l:state.termwinid = win_getid(winnr())
+    execute l:cd . fnameescape(l:dir)
+
+    " Adjust the window width or height depending on whether it's a vertical or
+    " horizontal split.
+    if l:mode =~ "vertical" || l:mode =~ "vsplit" || l:mode =~ "vnew"
+      exe 'vertical resize ' . l:width
+    elseif mode =~ "split" || mode =~ "new"
+      exe 'resize ' . l:height
+    endif
+    " we also need to resize the pty, so there you go...
+    call jobresize(l:state.id, l:width, l:height)
+
+  " setup term for vim8
+  elseif has('terminal')
+    let l:term = {
+          \ 'term_rows' : l:height,
+          \ 'term_cols' : l:width,
+          \ 'out_cb': function('s:out_cb', [], state),
+          \ 'exit_cb' : function('s:exit_cb', [], state),
+        \ }
+
+    if l:mode =~ "vertical" || l:mode =~ "vsplit" || l:mode =~ "vnew"
+      let l:term["vertical"] = 1
```

Comment:
When the user has used one of the modifiers (e.g. `belowright`), `l:term["vertical"] = 1` won't honor that modifier correctly. The docs say that modifiers are supported by `vertical`. Can you refactor to make sure this supports them correctly?

Reply:
@bhcleek 

Not sure I follow this correctly. That "vertical" is coming from the vim-go config file. Would a user ever have a chance to specify `belowright` when spawning a terminal via vim-go commands?

Reply:
The docs for `term_start`'s `vertical` parameter read
> split the window vertically; note that other window position can be defined with command modifiers, such as |:belowright|.

`g:go_term_mode` is the command that's used to open the terminal window. I suspect that `l:term["vertical"]`'s value should be `l:mode` so that `g:go_term_mode` is honored in Vim8 as well as it's honored in Neovim.